packages/mui-system/src/Grid/createGrid.tsx

```diff
@@ -0,0 +1,271 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { OverridableComponent } from '@mui/types';
+import {
+  unstable_composeClasses as composeClasses,
+  unstable_generateUtilityClass as generateUtilityClass,
+} from '@mui/utils';
+import systemStyled from '../styled';
+import useThemePropsSystem from '../useThemeProps';
+import useTheme from '../useTheme';
+import { extendSxProp } from '../styleFunctionSx';
+import createTheme from '../createTheme';
+import {
+  generateGridStyles,
+  generateGridSizeStyles,
+  generateGridColumnsStyles,
+  generateGridColumnSpacingStyles,
+  generateGridRowSpacingStyles,
+  generateGridDirectionStyles,
+  generateGridOffsetStyles,
+} from './gridGenerator';
+import { CreateMUIStyled } from '../createStyled';
+import { GridBaseProps, GridTypeMap } from './GridProps';
+
+const defaultTheme = createTheme();
+
+// widening Theme to any so that the consumer can own the theme structure.
+const defaultCreateStyledComponent = (systemStyled as CreateMUIStyled<any>)('div', {
+  name: 'MuiGrid',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+});
+
+function useThemePropsDefault<T>(props: T) {
+  return useThemePropsSystem({
+    props,
+    name: 'MuiGrid',
+    defaultTheme: { ...defaultTheme, components: {} },
+  });
+}
+
+function resolveSpacingClasses(
+  spacing: GridBaseProps['spacing'],
+  container?: boolean,
+  styles: Record<string, string> = {},
```

Comment:
```suggestion
  classes: Record<string, string> = {},
```

Isn't this the better word?